From: Michal Hocko <mhocko@suse.com>
Subject: disable prot_none native mitigation
Patch-mainline: never (suse specific)
References: bnc#1104684

This is a temporary workaround for code that doesn't expect mangled prot_none
entries. One known example is THP split path but there might be more and we
are not able to review all of them before CRD. This should be rechecked later
and patch dropped.

Signed-off-by: Michal Hocko <mhocko@suse.com>

---
 arch/x86/include/asm/pgtable-invert.h |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/arch/x86/include/asm/pgtable-invert.h
+++ b/arch/x86/include/asm/pgtable-invert.h
@@ -6,7 +6,14 @@
 
 static inline bool __pte_needs_invert(u64 val)
 {
+	/*
+	 * temporary workaround until we catch all unexpected usage of prot_none
+	 * entries
+	 */
+#if 0
 	return (val & (_PAGE_PRESENT|_PAGE_PROTNONE)) == _PAGE_PROTNONE;
+#endif
+	return false;
 }
 
 /* Get a mask to xor with the page table entry to get the correct pfn. */
